django/core/urlresolvers.py

```diff
@@ -37,12 +37,17 @@
 
 
 class ResolverMatch(object):
-    def __init__(self, func, args, kwargs, url_name=None, app_name=None, namespaces=None):
+    def __init__(self, func, args, kwargs, url_name=None, app_names=None, namespaces=None):
         self.func = func
         self.args = args
         self.kwargs = kwargs
         self.url_name = url_name
-        self.app_name = app_name
+
+        if app_names:
+            self.app_names = [x for x in app_names if x]
```

Comment:
A comment about why we need to filter out blank app_names would be useful. (e.g. how do blank names get there in the first place?) Personally I like a ternary for this:
`self.app_names = [x for x in app_names if x] if app_names else []`